resolvers.py

```diff
@@ -19,20 +19,35 @@ def get_dates_and_price_check(
             _,
             info,
             **search_criteria
-    ) -> List[DatesCheck]:
-        validated_search_criteria = DatesAndPriceCheckInputValidator(**search_criteria)
-        id_houses_dict = {}
-        for listing_id in validated_search_criteria.listing_ids:
-            id_houses_dict[listing_id] = data[listing_id]
-        return DatesCheckGetter.get_date_check_results(validated_search_criteria, id_houses_dict)
+    ) -> List[Union[DatesCheck, Error]]:
+        try:
+            validated_search_criteria = DatesAndPriceCheckInputValidator(**search_criteria)
+            dates_and_price_check_results = []
+            for listing_id in validated_search_criteria.listing_ids:
+                if listing_id in data.keys():
+                    dates_and_price_check_results.append(
+                        DatesCheckGetter.get_date_check_result_for_house(
+                            validated_search_criteria,
+                            data[listing_id]
+                        )
+                    )
+                else:
+                    dates_and_price_check_results.append(ListingUnsupportedError(f"Listing {listing_id} out of scope"))
+
+            return dates_and_price_check_results
+
+        except Exception as e:
+            return [GenericError(f"Something seems to have gone wrong. Please check the error message {e}")]
 
     @staticmethod
     @query.field("ranked_search_results")
     def get_ranked_search_results(
             _,
             info,
             **search_criteria
-    ) -> List[DatesCheck]:
-        validated_search_criteria = RankedSearchResultInputValidator(**search_criteria)
-        print(type(validated_search_criteria))
-        return DatesCheckGetter.get_date_check_results(validated_search_criteria, data)
+    ) -> List[Union[DatesCheck, Error]]:
```

Comment:
Update this.
I suggest in `graphql_objects.py` you define
```python
DatesResults = Union[DatesCheck, ListingUnsupportedError, GenericError]
```
Then over here you just do `List[DatesResult]`

Please check for other occurrences of this.